### PR TITLE
Allow the Gateway controller to optionally process Gateway CRDs from any watched namespace

### PR DESCRIPTION
**Pull request by GrahamGoudeau**
The Gateway controller can now optionally process Gateway CRDs from anywhere in the watch namespaces, not just the namespace where the controller is running. This is accomplished by adding a Helm values field "gateway.readGatewaysFromAllNamespaces". This new field defaults to false to preserve the old Gateway controller behavior, which is to read Gateway CRDs only from the namespace in which the controller is running. When the field is set to true, it will read Gateway CRDs from any of the watch namespaces.
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1489

### CODE REVIEW

**Comment by kdorosh** [comment]
/kick nil flake https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=1bb0be66-1fe3-48cf-a5de-ded8b61aec1a

File: changelog/v1.0.0-rc3/cross-namespace-gateways.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+changelog:
+  - type: NEW_FEATURE
+    issueLink: https://github.com/solo-io/gloo/issues/1489
+    description: >
+      The Gateway controller can now optionally process Gateway CRDs from anywhere in the watch
+      namespaces, not just the namespace where the controller is running. The new setting
+      controlling that can be found in the Gloo Settings object, at
+      "gateway.readGatewaysFromAllNamespaces". The value of this field can be configured at install
```
Comment (marcogschmidt) [inline changelog/v1.0.0-rc3/cross-namespace-gateways.yaml]: Nit:
```suggestion
      `gateway.readGatewaysFromAllNamespaces`. The value of this field can be configured at install
```

File: changelog/v1.0.0-rc3/cross-namespace-gateways.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      time by setting the new Helm field "gateway.readGatewaysFromAllNamespaces" to your desired
```
Comment (marcogschmidt) [inline changelog/v1.0.0-rc3/cross-namespace-gateways.yaml]: Same nit:
```suggestion
      time by setting the new Helm field `gateway.readGatewaysFromAllNamespaces` to your desired
```

File: changelog/v1.0.0-rc3/cross-namespace-gateways.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      value; it defaults to false to preserve the old behavior.
+  - type: HELM
+    issueLink: https://github.com/solo-io/gloo/issues/1489
+    description: >
+      Add a Helm values field "gateway.readGatewaysFromAllNamespaces". This new field defaults to false
```
Comment (marcogschmidt) [inline changelog/v1.0.0-rc3/cross-namespace-gateways.yaml]:
```suggestion
      Add a Helm values field `gateway.readGatewaysFromAllNamespaces`. This new field defaults to `false`
```

File: changelog/v1.0.0-rc3/cross-namespace-gateways.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      to preserve the old Gateway controller behavior, which is to read Gateway CRDs only from the
+      namespace in which the controller is running. When the field is set to true, it will read Gateway
```
Comment (marcogschmidt) [inline changelog/v1.0.0-rc3/cross-namespace-gateways.yaml]:
```suggestion
      namespace in which the controller is running. When the field is set to `true`, it will read Gateway
```

File: projects/gloo/api/v1/settings.proto [hunk 5]
```diff
@@ -383,4 +383,8 @@ message GatewayOptions {
     // if provided, the Gateway will perform[Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
     // of Gateways, Virtual Services, and Route Tables when running in Kubernetes.
     ValidationOptions validation = 3;
+
+    // when true, the Gateway controller will consume Gateway CRDs from all watch namespaces, rather
```
Comment (marcogschmidt) [inline projects/gloo/api/v1/settings.proto]:
```suggestion
    // When true, the Gateway controller will consume Gateway CRDs from all watch namespaces, rather
```

File: projects/gloo/api/v1/settings.proto [hunk 6]
```diff
@@ -383,4 +383,8 @@ message GatewayOptions {
     // if provided, the Gateway will perform[Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
```
Comment (marcogschmidt) [inline projects/gloo/api/v1/settings.proto]: Not part of this PR,but can you fix this as well?
```suggestion
    // If provided, the Gateway will perform[Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
```

File: projects/gateway/pkg/translator/translator.go [hunk 7]
```diff
@@ -119,10 +123,12 @@ func gatewaysRefsToString(gateways v1.GatewayList) []string {
 // https://github.com/solo-io/gloo/issues/538
 // Gloo should only pay attention to gateways it creates, i.e. in it's write namespace, to support
 // handling multiple gloo installations
-func filterGatewaysForNamespace(gateways v1.GatewayList, namespace string) v1.GatewayList {
+func (t *translator) filterGatewaysForNamespace(gateways v1.GatewayList, namespace string) v1.GatewayList {
 	var filteredGateways v1.GatewayList
 	for _, gateway := range gateways {
-		if gateway.Metadata.Namespace == namespace {
+		// https://github.com/solo-io/gloo/issues/1489
+		// we may want to configure the controller to read all the Gateway CRDs it can find
```
Comment (marcogschmidt) [inline projects/gateway/pkg/translator/translator.go]: Remove issue link?

File: projects/gateway/pkg/translator/translator_test.go [hunk 8]
```diff
@@ -199,6 +200,73 @@ var _ = Describe("Translator", func() {
...
+										}},
+									},
+								},
+							},
+						},
+					},
+				}
+			})
+
+			It("should translate proxy with default name", func() {
```
Comment (marcogschmidt) [inline projects/gateway/pkg/translator/translator_test.go]: Probably a copy-paste leftover, please update.
